xarray/tests/test_dataset.py

```diff
@@ -4035,6 +4035,26 @@ def data_set(seed=None):
     return create_test_data(seed)
 
 
+@pytest.mark.parametrize('unaligned_coords', (
+    {'x': [2, 1, 0]},
+    {'x': (['x'], np.asarray([2, 1, 0]))},
+    {'x': (['x'], np.asarray([1, 2, 0]))},
+    {'x': pd.Index([2, 1, 0])},
+    {'x': Variable(dims='x', data=[0, 2, 1])},
```

Comment:
Can you add test cases where there are extra `unaligned_coords` that don't correspond to an index?

Examples:
- Scalar coordinates (e.g., `{'y': 42}`)
- 1D coordinates that don't match dimensions (e.g., `{'y': ('x', [2, 1, 0])}`)
- 1D coordinates along the wrong dimension (e.g., `{'x': ('y', [2, 1, 0])}`). These should still raise with an informative error message (I think this will still work, but it's good to double check).